pandas/tests/test_strings.py

```diff
@@ -435,6 +435,12 @@ def test_replace(self):
                 for data in (['a', 'b', None], ['a', 'b', 'c', 'ad']):
                     values = klass(data)
                     self.assertRaises(TypeError, values.str.replace, 'a', repl)
+        
+        # GH 15055, callable repl
+        repl = lambda m: m.group(0).swapcase()
+        result = values.str.replace('[a-z][A-Z]{2}', repl, n=2)
+        exp = Series([u('foObaD__baRbaD'), NA])
```

Comment:
With `inspect` I could do

```py
n_empty_args = sum(1 for p in inspect.signature(repl).parameters.values() 
                   if p.kind in (p.POSITIONAL_ONLY, p.POSITIONAL_OR_KEYWORD, p.KEYWORD_ONLY) 
                   and p.default is p.empty)

# or better readable syntax

signature = inspect.signature(repl)
empty_args = [p for p in signature.parameters.values() 
              if p.kind in (p.POSITIONAL_ONLY, p.POSITIONAL_OR_KEYWORD, p.KEYWORD_ONLY) 
              and p.default is p.empty]
n_empty_args = len(empty_args)

# followed by

if n_empty_args > 1:
    raise TypeError("callable should have at most one positional without a default")
```

`n_empty_args != 1` might work, unless the first arg has a default for some reason.